### PR TITLE
AYON: Addons creation enhancements

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Changelog Description
Enhanced AYON addons creation. Fix issue with `Pattern` typehint. Zip filenames contain version. OpenPype package is skipping modules that are already separated in AYON. Updated settings of addons.

## Additional information
Created openpype package does not contain ftrack, shotgrid, sync server and slack modules.

## Testing notes:
1. Create packages
2. Upload them to server
3. Run ayon-launcher and validate they still do work

### CODE REVIEW

File: server_addon/blender/server/settings/publish_plugins.py [hunk 1]
```diff
@@ -138,12 +142,13 @@ class PublishPuginsModel(BaseSettingsModel):
         "enabled": True,
         "optional": True,
         "active": True,
-        "families": [
+        "families":  [
```
[the remark this answers is not in the document]
Comment (iLLiCiTiT) [reply server_addon/blender/server/settings/publish_plugins.py:145]:
```suggestion
        "families": [
```

**Review by mkolar — APPROVED** [review]
Looking good, builds and install correctly too. 

I have a couple of settings issues, but they seem unrelated.

**Review by mkolar — CHANGES_REQUESTED** [review]
Actually

```
Traceback (most recent call last):
  File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\tools\launcher\widgets.py", line 238, in _on_projects_refresh
    self.discover_actions()
  File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\tools\launcher\widgets.py", line 227, in discover_actions
    self.model.discover()
  File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\tools\launcher\models.py", line 81, in discover
    app_actions = self.get_application_actions()
  File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\tools\launcher\models.py", line 98, in get_application_actions
    project_settings = get_project_settings(project_name)
  File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\settings\lib.py", line 1144, in get_project_settings
    return get_ayon_project_settings(default_settings, project_name)
  File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\settings\ayon_settings.py", line 1377, in get_ayon_project_settings
    return convert_project_settings(ayon_settings, default_values)
  File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\settings\ayon_settings.py", line 1254, in convert_project_settings
    _convert_nuke_project_settings(ayon_settings, output)
  File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\settings\ayon_settings.py", line 728, in _convert_nuke_project_settings
    item["reformat_node_config"])
KeyError: 'reformat_node_config'
```
No Applications loading in launcher with nuke 0.1.1

**Comment by iLLiCiTiT** [comment]
> Actually
> 
> ```
> Traceback (most recent call last):
>   File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\tools\launcher\widgets.py", line 238, in _on_projects_refresh
>     self.discover_actions()
>   File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\tools\launcher\widgets.py", line 227, in discover_actions
>     self.model.discover()
>   File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\tools\launcher\models.py", line 81, in discover
>     app_actions = self.get_application_actions()
>   File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\tools\launcher\models.py", line 98, in get_application_actions
>     project_settings = get_project_settings(project_name)
>   File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\settings\lib.py", line 1144, in get_project_settings
>     return get_ayon_project_settings(default_settings, project_name)
>   File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\settings\ayon_settings.py", line 1377, in get_ayon_project_settings
>     return convert_project_settings(ayon_settings, default_values)
>   File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\settings\ayon_settings.py", line 1254, in convert_project_settings
>     _convert_nuke_project_settings(ayon_settings, output)
>   File "C:\Users\milan\AppData\Local\Ynput\AYON\addons\openpype_3.16.3-nightly.2\openpype\settings\ayon_settings.py", line 728, in _convert_nuke_project_settings
>     item["reformat_node_config"])
> KeyError: 'reformat_node_config'
> ```
> 
> No Applications loading in launcher with nuke 0.1.1

This meants you've used different openpype addon than what this PR is offering.

**Review by mkolar — APPROVED** [review]
all good now

**Pull request merged**